src/ZkMinterERC1155V1.sol

```diff
@@ -0,0 +1,186 @@
+  error ZkMinterERC1155V1__InsufficientBalance();
+
+  /*///////////////////////////////////////////////////////////////
+                          State Variables
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice The ERC1155 contract used to verify caller eligibility for minting.
+  IERC1155 public erc1155;
+
+  /// @notice The specific token ID within the ERC1155 contract whose balance is checked.
+  uint256 public tokenId;
+
+  /// @notice The minimum balance of `tokenId` the caller must hold to mint.
+  uint256 public balanceThreshold;
+
+  /*///////////////////////////////////////////////////////////////
+                          Constructor
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice Initializes the contract with the mintable contract, admin, erc1155 contract, token id, and balance
+  /// threshold.
+  /// @param _mintable A contract used as a target when calling mint. Any contract that conforms to the IMintable
+  /// interface can be used, but in most cases this will be another `ZKMinter` extension or `ZKCappedMinter`.
+  /// @param _admin The address that will have admin privileges.
+  /// @param _erc1155 The ERC1155 contract to use for minting.
+  /// @param _tokenId The token id for the ERC1155 contract.
+  /// @param _balanceThreshold The balance threshold for the ERC1155 contract.
+  constructor(IMintable _mintable, address _admin, address _erc1155, uint256 _tokenId, uint256 _balanceThreshold) {
+    if (_admin == address(0)) {
+      revert ZkMinterERC1155V1__InvalidZeroAddress();
+    }
+
+    _updateMintable(_mintable);
+    _updateERC1155(_erc1155);
+    _updateTokenId(_tokenId);
+    _updateBalanceThreshold(_balanceThreshold);
+
+    _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+    _grantRole(PAUSER_ROLE, _admin);
+  }
+
+  /*///////////////////////////////////////////////////////////////
+                          External Functions
+  //////////////////////////////////////////////////////////////*/
+
+  /// @notice Allows a caller to mint tokens to a specified address if they meet the ERC1155 balance threshold.
+  /// @param _to The address to mint the tokens to.
+  /// @param _amount The amount of tokens to mint.
+  /// @dev Caller must hold at least `balanceThreshold` of token `tokenId` in `erc1155`.
+  function mint(address _to, uint256 _amount) external {
+    _revertIfClosed();
+    _requireNotPaused();
+    _revertIfInsufficientBalance();
+
+    if (_to == address(0)) {
+      revert ZkMinterERC1155V1__InvalidZeroAddress();
+    }
+
+    if (_amount == 0) {
+      revert ZkMinterERC1155V1__InvalidAmount();
+    }
+
+    // mint the tokens
+    mintable.mint(_to, _amount);
+
+    emit Minted(msg.sender, _to, _amount);
+  }
+
+  /// @notice Updates the ERC1155 contract for the ZkMinter.
+  /// @param _erc1155 The ERC1155 contract to use for balance checks.
+  function updateERC1155(address _erc1155) external {
```

Comment:
@alexkeating Not in spec but since we can update token id and threshold. Figured this might be useful. What's your take on this